Congrats: remove legacy thank you code

chad1008 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 6015-gh-Automattic/dotcom-forge

## Proposed Changes

Remove legacy unused/unreachable thank you code from [client/my-sites/checkout/checkout-thank-you/index.tsx](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.tsx)

There is additional code to remove in other files that will be split up/batched into followup(s).

## Testing Instructions

Confirm that these purchase flows continue to match behavior on `trunk`:

- Domain only
- Dlan only
- Ecomm plan only
- Plan with domain 
- Add-ons (e.g. Jetpack AI)
- Delayed domain transfer (i.e. select a domain transfer during signup)
- Titan email
- Google Workspace email
- Bulk domain transfer (e.g. https://wordpress.com/transfer-google-domains/)

All but Delayed domain transfers should render the new Congrats pages.

chad1008 committed:
remove domain and default thanks page

chad1008 committed:
remove legacy subcomponents and logic

chad1008 committed:
restore `isWooCommerceInstalled`

stephanethomas commented:
It appears that both unit tests and E2E tests have failed. More importantly, this pull request removes a significant amount of code, some of which including conditional branches, which makes it difficult for me to be confident that we're ditching the right things. I'm reluctant to test these changes against each mentioned flow, as it will be not only time-consuming but also risky, given that we might miss a specific flow.

I'd like to suggest a different approach where we record a Tracks event at the beginning of each major block of code we plan to remove, and let that run for a week or two. This will allow us to confirm that these code paths are indeed no longer in use. What do you think? Or perhaps I'm being too risk-averse 🙂?

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -684,72 +624,7 @@ export class CheckoutThankYou extends Component<
 					<PlanThankYouCard siteId={ this.props.selectedSite?.ID ?? 0 } { ...planProps } />
 				</Main>
 			);
-		} else if ( wasDomainProduct && ! wasBulkDomainTransfer ) {
```
Noting that, [`isRefactoredForThankYouV2()`](https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts#L11) returns `true` for all scenarios except for a failed order or delayed transfer are considered redesigned. 

So domain-only purchases will enter the [refactored logic](https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L590), and the page will be rendered by: https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L604-L605

The block of condition starting at L687 is never executed.

wongasy commented:
>both unit tests and E2E tests have failed. 

Those tests have been failing since yesterday (see p1712772550012179-slack-C02DQP0FP), but it should be fixed by https://github.com/Automattic/wp-calypso/pull/89420. A rebase should address the issue in this PR!

As for the unit tests, @chad1008, you might need to remove tests that are no longer relevant.

> More importantly, this pull request removes a significant amount of code, some of which including conditional branches, which makes it difficult for me to be confident that we're ditching the right things. I'm reluctant to test these changes against each mentioned flow, as it will be not only time-consuming but also risky, given that we might miss a specific flow.

@stephanethomas I have full context on these changes and am fairly confident about removing the code. I'll take over the review :) I've also added inline comments to explain why we're removing the code.

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -684,72 +624,7 @@ export class CheckoutThankYou extends Component<
-			return null;
-		}
-
-		// standard thanks page
-		return (
-			<Main className="checkout-thank-you">
-				<PageViewTracker { ...this.getAnalyticsProperties() } title="Checkout Thank You" />
-				<Card className="checkout-thank-you__content">{ this.productRelatedMessages() }</Card>
-				{ showHappinessSupport && (
-					<Card className="checkout-thank-you__footer">
-						<HappinessSupport
-							isJetpack={ wasJetpackPlanPurchased }
-							contactButtonEventName="calypso_plans_autoconfig_chat_initiated"
-						/>
-					</Card>
-				) }
-			</Main>
-		);
```
@chad1008 We might want to keep this for handling failed purchases.

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -768,169 +643,6 @@ export class CheckoutThankYou extends Component<
-		return [];
-	};
-
-	productRelatedMessages = () => {
```
Since we're keeping the legacy standard thank you page (see previous comment), this function should be kept.

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -768,169 +643,6 @@ export class CheckoutThankYou extends Component<
-		const componentAndPrimaryPurchaseAndDomain = this.getComponentAndPrimaryPurchaseAndDomain();
-		const [ component, primaryPurchase ] = componentAndPrimaryPurchaseAndDomain;
-
-		if ( ! this.isDataLoaded() ) {
-			return (
-				<div>
-					<CheckoutThankYouHeader
-						isDataLoaded={ false }
-						isSimplified={ isSimplified }
-						selectedSite={ selectedSite }
-						upgradeIntent={ upgradeIntent }
-						siteUnlaunchedBeforeUpgrade={ siteUnlaunchedBeforeUpgrade }
-						displayMode={ displayMode }
-					/>
-
-					{ ! isSimplified && (
-						<>
-							<CheckoutThankYouFeaturesHeader isDataLoaded={ false } />
-
-							<div className="checkout-thank-you__purchase-details-list">
-								<PurchaseDetail isPlaceholder />
-								<PurchaseDetail isPlaceholder />
-								<PurchaseDetail isPlaceholder />
-							</div>
-						</>
-					) }
-				</div>
-			);
-		}
```
Noting that this is the placeholder that @nightnei will be updating to align with the redesigned congrats page as part of this issue https://github.com/Automattic/dotcom-forge/issues/6500

nightnei replied:
If we go with my approach - we can remove it. Details in my PR with the proposed changes https://github.com/Automattic/wp-calypso/pull/89487#discussion_r1562697973

chad1008 replied:
Excellent, thanks for cross linking... I'll watch your PR and plan on rebasing/updating as needed when yours merges @nightnei

wongasy replied:
@chad1008 Initially, I thought the placeholder PR will be merged first, but that didn't turn out to be the case. If you're able to address the final feedback and get this PR shipped today, then Vova can rebase his PR :)

chad1008 replied:
Just finished now and was about to confirm whether or not Vova's PR also removed this placeholder 😄

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -768,169 +643,6 @@ export class CheckoutThankYou extends Component<
-	 * returns an array of varying size with the component instance,
-	 * then an optional purchase object possibly followed by a domain name
-	 */
-	getComponentAndPrimaryPurchaseAndDomain = (): ComponentAndPrimaryPurchaseAndDomain => {
```
We should also keep this function, but remove all conditions except for failed purchases. 

Congrats page for plans is handled by the `<PlanOnlyThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L607-L613

Congrats page for Google Workspace is handled by the `<GoogleWorkspaceSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L636-L639

Congrats page for Titan is handled by the `<TitanSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L630-L635

Congrats page for all other purchases will be handled by the `<GenericThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L641

wongasy commented on client/my-sites/checkout/checkout-thank-you/index.tsx:
```diff
@@ -989,130 +701,3 @@ export default connect(
- * Retrieves the component (and any corresponding data) that should be displayed according to the type of purchase
- * just performed by the user.
- */
-function PurchaseDetailsWrapper(
```
Similarly, this function should be kept, but with all conditions except for `hasFailedPurchases` removed.

chad1008 committed:
restore code needed for failed purchases

chad1008 commented:
Thanks for the reviews (both of them!)

@wongasy I believe I've addressed your feedback regarding the functions we'll want to keep, but would love a second pair of eyes. I'm going to let CI run and then look into tracking down any obsolete unit tests.

wongasy reviewed (commented):
We're almost there! Once the following issues are addressed, we should be good to go!

* The white background is gone for the Generic Congrats Page. That can be addressed with this fix: https://github.com/Automattic/wp-calypso/pull/88359/files#diff-ca8bf6f94e9100bf77a11d24ea3c90dc5ccffd14427a6efac01280f20b4c0e2aR666

<img width="1002" alt="Screen Shot 2024-04-15 at 1 50 30 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/a8f236f3-c272-43fd-9553-9bdce88362f6">

* There's a bug in production where the "Need extra help" link is hugging the edge of the page. Let's take the opportunity to fix that!

<img width="1061" alt="Screen Shot 2024-04-15 at 2 06 38 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/a137356c-36cb-4772-8a11-32e0593c0856">

* There's another a bug in production where we have extra padding on mobile. Let's also take the opportunity to fix that!

<img width="459" alt="Screen Shot 2024-04-15 at 2 12 54 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/79450bc7-c752-4485-b2c7-ae02fcc508b3">

The last two issues should be addressed with the following diff:

```diff
--- a/client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx
+++ b/client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx
@@ -8,6 +8,7 @@ import { DefaultMasterbarContact } from './default-contact';
 const MasterbarStyledBlock = styled( Masterbar )`
        --color-masterbar-background: var( --studio-white );
        --color-masterbar-text: var( --studio-gray-60 );
+       padding-right: 24px;
        border-bottom: 0;
 `;
 
diff --git a/client/my-sites/checkout/checkout-thank-you/style.scss b/client/my-sites/checkout/checkout-thank-you/style.scss
index e504af752d..fd09910f8b 100644
--- a/client/my-sites/checkout/checkout-thank-you/style.scss
+++ b/client/my-sites/checkout/checkout-thank-you/style.scss
@@ -28,7 +28,7 @@
 
                &.main {
                        max-width: 750px;
-                       padding: 0 24px 0;
+                       padding: 0;
                        display: flex;
                        flex-direction: column;
                        text-align: initial;
@@ -663,6 +663,15 @@
        }
 }
```

chad1008 committed:
css fixes

chad1008 committed:
remove legacy placeholder

chad1008 commented:
I've applied the CSS fixes and removed that legacy placeholder that @nightnei's PR will be replacing, so when that one is rebased it won't be there cluttering things up any longer.

One more review and I think we'll be in good shape! cc @wongasy

wongasy commented:
>I've applied the CSS fixes and removed that legacy placeholder that @nightnei's PR will be replacing, so when that one is rebased it won't be there cluttering things up any longer.

@chad1008 We should not remove that placeholder, if the idea is to ship this PR before Vova's 😅

chad1008 committed:
Revert "remove legacy placeholder"

This reverts commit 64e6996a97aab0e53e5d7b94bfa55cab445c1e3b.

chad1008 commented:
> We should not remove that placeholder, if the idea is to ship this PR before Vova's 😅 Otherwise, we'll be left with no placeholder.

🤦 yeah, let's pretend I didn't say any of that please

chad1008 commented:
current e2e failures look unrelated, and have been failing consistently on other PRs for the last half hour or so. I'll check back periodically and re-run until we get a green light

wongasy reviewed (approved):
LGTM and tests well!

I've tested the Congrats page for the following purchases:

* Domain only

<img width="300" alt="Screen Shot 2024-04-15 at 8 40 22 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/203a57bc-94bf-4869-a36b-466a91ab6344">

* Domain + Email + Plan

<img width="300" alt="Screen Shot 2024-04-15 at 7 42 42 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/45c28736-df70-4476-8b95-a950ba725284">

* Domain + Titan

<img width="300" alt="Screen Shot 2024-04-15 at 7 53 07 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/13867d9f-0056-4f75-b099-b73cc10bff66">

* Multiple Titan mailboxes

<img width="300" alt="Screen Shot 2024-04-15 at 7 43 20 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/f0e3ad30-44b3-42a5-8806-06f162b248d6">

* Google Workspace

<img width="300" alt="Screen Shot 2024-04-15 at 7 43 06 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/6f92f3cf-a8d6-457b-8e60-36259a19732b">

* Google Workspace additional mailbox

<img width="300" alt="Screen Shot 2024-04-15 at 7 42 55 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/8560926d-19b4-468d-9a36-67ab52fead0a">

* E-Commerce plan 

<img width="300" alt="Screen Shot 2024-04-15 at 7 44 57 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/50d229fb-72b2-47d5-a454-710994b26b3a">

* Domain transfer

<img width="300" alt="Screen Shot 2024-04-15 at 7 46 33 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/f21a89bd-4147-464a-a935-6f36b98f92ec">

wongasy commented:
>current e2e failures look unrelated

Hm... I've flagged it in the #calypso channel p1713226175237849-slack-C02DQP0FP
